### PR TITLE
Refactor DDLStatement to MySQL, Oracle, PostgreSQL, SQL92, SQLServer DDLStatement

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #7170 .

Changes proposed in this pull request:
- Refactor DDLStatement to MySQL, Oracle, PostgreSQL, SQL92, SQLServer DDLStatement

### CODE REVIEW

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi, Resolving conflicts become the first step for my review.

**Comment by strongduanmu** [comment]
> Hi, Resolving conflicts become the first step for my review.

@tristaZero I am very sorry, and the conflict has been resolved now.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java [hunk 1]
```diff
@@ -22,5 +22,5 @@
 /**
  * Drop function statement.
  */
-public final class DropFunctionStatement extends AbstractSQLStatement implements DDLStatement {
+public abstract class DropFunctionStatement extends AbstractSQLStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java:25]: I saw none of `xxx DropFunctionStatement` and this class has members, so what role does `DropFunctionStatement` play?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java:25]: > I saw none of `xxx DropFunctionStatement` and this class has members, so what role does `DropFunctionStatement` play?

@tristaZero `DropFunctionStatement` has two implementation classes——`MySQLDropFunctionStatement` and `PostgreSQLDropFunctionStatement`, and they have been included in PR.

image

image

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java:25]: Why do all of them have no members?  `MySQLDropFunctionStatement` and `PostgreSQLDropFunctionStatement ` will be filled with member variables later?

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropFunctionStatement.java:25]: > Why do all of them have no members? `MySQLDropFunctionStatement` and `PostgreSQLDropFunctionStatement ` will be filled with member variables later?

@tristaZero Yes, they will be optimized in #6498. Later, the table name in the statement will be parsed and the table will be verified.

**Review by tristaZero — APPROVED** [review]

**Pull request merged**